src/renderer/base/RenderSettings.cpp

```diff
@@ -196,74 +156,37 @@ std::pair<COLORREF, COLORREF> RenderSettings::GetAttributeColors(const TextAttri
-        auto fgIndex = fgTextColor.IsDefault() ? AdjustedFgIndex : fgTextColor.GetIndex();
-
-        if (brightenFg)
-        {
-            // There is a special case for intense here - we need to get the bright version of the foreground color
-            if (fgTextColor.IsIndex16() && (fgIndex < 8))
-            {
-                fgIndex += 8;
-            }
-            else if (fgTextColor.IsDefault())
-            {
-                fgIndex = AdjustedBrightFgIndex;
-            }
-        }
+    auto fg = fgTextColor.GetColor(_colorTable, defaultFgIndex, brightenFg);
+    auto bg = bgTextColor.GetColor(_colorTable, defaultBgIndex);
 
-        if (swapFgAndBg)
-        {
-            const auto fg = _adjustedForegroundColors[fgIndex][bgIndex];
-            const auto bg = fgTextColor.GetColor(_colorTable, defaultFgIndex, brightenFg);
-            return { fg, bg };
-        }
-        else
-        {
-            const auto fg = _adjustedForegroundColors[bgIndex][fgIndex];
-            const auto bg = bgTextColor.GetColor(_colorTable, defaultBgIndex);
-            return { fg, bg };
-        }
+    if (dimFg)
+    {
+        fg = (fg >> 1) & 0x7F7F7F; // Divide foreground color components by two.
     }
-    else
+    if (swapFgAndBg)
     {
-        auto fg = fgTextColor.GetColor(_colorTable, defaultFgIndex, brightenFg);
-        auto bg = bgTextColor.GetColor(_colorTable, defaultBgIndex);
-
-        if (dimFg)
-        {
-            fg = (fg >> 1) & 0x7F7F7F; // Divide foreground color components by two.
-        }
-        if (swapFgAndBg)
-        {
-            std::swap(fg, bg);
-        }
-        if (attr.IsInvisible())
-        {
-            fg = bg;
-        }
+        std::swap(fg, bg);
+    }
+    if (attr.IsInvisible())
+    {
+        fg = bg;
+    }
 
-        // We intentionally aren't _only_ checking for attr.IsInvisible here, because we also want to
-        // catch the cases where the fg was intentionally set to be the same as the bg. In either case,
-        // don't adjust the foreground.
-        if (Feature_AdjustIndistinguishableText::IsEnabled() &&
+    // We intentionally aren't _only_ checking for attr.IsInvisible here, because we also want to
+    // catch the cases where the fg was intentionally set to be the same as the bg. In either case,
+    // don't adjust the foreground.
+    if constexpr (Feature_AdjustIndistinguishableText::IsEnabled())
+    {
+        if (
+            _renderMode.any(Mode::IndexedDistinguishableColors, Mode::AlwaysDistinguishableColors) &&
             fg != bg &&
-            GetRenderMode(Mode::AlwaysDistinguishableColors))
+            (_renderMode.test(Mode::AlwaysDistinguishableColors) || (fgTextColor.IsDefaultOrIndex16() && bgTextColor.IsDefaultOrIndex16())))
         {
-            fg = ColorFix::GetPerceivableColor(fg, bg);
+            fg = ColorFix::GetPerceivableColor(fg, bg, 0.5f * 0.5f);
         }

```

Comment:
Note that there's a slight change in behavior here from the previous implementation (I think). When you don't have the `AlwaysDistinguishableColors` mode set, it would previously not have applied to dimmed colors, but it will do now (assuming I'm reading this correctly). I'm not sure whether anyone would care either way, but I thought it worth mentioning.

Reply:
Oh wow, I totally forgot to mention that in this PR!

I did it intentionally. Since `GetPerceivableColor` could be considered a very important accessibility feature, dim colors need to be just as visible, as non-dim colors are, for people relying on this feature. Since the new `GetPerceivableColor` function is fairly "accurate" at not changing the lightness more than necessary, the result still looks quite okay to me.